acme/errors.go

```diff
@@ -293,16 +293,31 @@ type Subproblem struct {
 	Identifier *Identifier `json:"identifier,omitempty"`
 }
 
+// NewError creates a new Error.
+func NewError(pt ProblemType, msg string, args ...interface{}) *Error {
+	return newError(pt, errors.Errorf(msg, args...))
+}
```

Comment:
Should this create a detailed error by default? We can also have a second constructor instead of WithAdditionalErrorDetail.

Reply:
Maybe it makes sense to do so, but we would have to check all paths that return an `Error` to not return sensitive details. Not impossible, but not the thing I was trying to solve in this PR.

Reply:
Something like NewDetailedError will be a new function. We can get replace just the ones with `WithAdditionalErrorDetail`